daemon/emer-daemon.c

```diff
@@ -1123,6 +1142,379 @@ set_max_bytes_buffered (EmerDaemon *self,
+  g_autoptr(GFileEnumerator) enumerator = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GFile) tallies_folder = NULL;
+  g_autofree gchar *tallies_path = NULL;
+  g_autofree gchar *date_string = NULL;
+
+  date_string = g_date_time_format (priv->current_aggregate_tally_date,
+                                    "%Y-%m-%d");
+  tallies_path = g_build_filename (priv->persistent_cache_directory,
+                                   "aggregate-timers",
+                                   date_string,
+                                   NULL);
+
+  tallies_folder = g_file_new_for_path (tallies_path);
+
+  enumerator = g_file_enumerate_children (tallies_folder,
+                                          "standard::*",
+                                          G_FILE_QUERY_INFO_NONE,
+                                          NULL,
+                                          &error);
+
+  if (error)
+    {
+      if (!g_error_matches (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+        g_warning ("Error listing tallies from %s: %s",
+                   tallies_path, error->message);
+      return;
+    }
+
+  while (TRUE)
+    {
+      g_autoptr(GVariant) variant = NULL;
+      g_autoptr(GVariant) cache_key = NULL;
+      g_autoptr(GVariant) event_id = NULL;
+      g_autoptr(GVariant) aggregate_key = NULL;
+      g_autoptr(GVariant) payload = NULL;
+      g_autoptr(GError) local_error = NULL;
+      g_autofree gchar *contents = NULL;
+      g_autofree gchar *path = NULL;
+      GFileInfo *info;
+      GFile *file;
+      gboolean complete;
+      guint32 counter;
+      uid_t unix_user_id;
+
+      g_file_enumerator_iterate (enumerator, &info, &file, NULL, &local_error);
+
+      if (local_error)
+        {
+          if (!g_error_matches (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+            g_warning ("Error listing tallies from %s: %s",
+                       tallies_path, local_error->message);
+          return;
+        }
+
+      if (!file || !info)
+        break;
+
+      if (g_file_info_get_file_type (info) != G_FILE_TYPE_REGULAR)
+        continue;
+
+      path = g_file_get_path (file);
+      g_file_get_contents (path, &contents, NULL, &local_error);
+
+      if (local_error)
+        {
+          g_warning ("Error loading tally from file %s: %s",
+                     path, local_error->message);
+          continue;
+        }
+
+      variant = g_variant_parse (NULL, contents, NULL, NULL, &local_error);
+      if (local_error)
+        {
+          g_warning ("Error parsing tally from file %s: %s",
+                     path, local_error->message);
+          continue;
+        }
+
+      g_variant_get (variant,
+                     "(vvvuubmv)",
+                     &event_id,
+                     &aggregate_key,
+                     &cache_key,
+                     &unix_user_id,
+                     &counter,
+                     &complete,
+                     &payload);
+
+      if (!complete)
+        continue;
```

Comment:
I don't think we need this field. We know that if the date is before today (this month for the monthly version), the event is complete and can be submitted to the server; and if it's today (or further into the future), it's not.

Reply:
What would happen on timezone or wallclock changes? The reason I kept it as a boolean field was to not implode on time travels backwards in time.

Reply:
We could instead delete data from this cache once it is enqueued for submission to the server. This would risk double-counting users if the clock moves backwards to the previous day but avoids the risk of double-counting time, I think.